File: src/main/java/kroppeb/server/command/arguments/IntRange.java
/*
 * Copyright (c) 2020 Kroppeb
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package kroppeb.server.command.arguments;

import kroppeb.server.command.reader.Reader;
import kroppeb.server.command.reader.ReaderException;

public class IntRange {
	final int minValue;
	final int maxValue;
	
	public IntRange(int minValue, int maxValue) {
		this.minValue = minValue;
		this.maxValue = maxValue;
	}
	
	public static IntRange read(Reader reader) throws ReaderException{
		int minValue;
		int maxValue;
		
		if(reader.tryRead("..")){
			minValue = Integer.MIN_VALUE;
			maxValue = reader.readInt();
		}else{
			minValue = reader.readInt();
			if(reader.tryRead("..")){
				if(reader.canRead() && !reader.isWhiteSpace()) {
					maxValue = reader.readInt();
				}else{
					maxValue = Integer.MAX_VALUE;
				}
			}else{
				maxValue = minValue;
			}
		}
		return new IntRange(minValue,maxValue);
	}
}

File: src/main/kotlin/kroppeb/server/command/arguments/IntRange.kt
/*
 * Copyright (c) 2020 Kroppeb
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
package kroppeb.server.command.arguments

import kroppeb.server.command.reader.Reader
import kroppeb.server.command.reader.ReaderException

class IntRange(val minValue: Int, val maxValue: Int) {

	companion object {
		@Throws(ReaderException::class)
		fun read(reader: Reader): IntRange {
			val minValue: Int
			val maxValue: Int
			if (reader.tryRead("..")) {
				minValue = Int.MIN_VALUE
				maxValue = reader.readInt()
			} else {
				minValue = reader.readInt()
				maxValue = if (reader.tryRead("..")) {
					if (reader.canRead() && !reader.isWhiteSpace) {
						reader.readInt()
					} else {
						Int.MAX_VALUE
					}
				} else {
					minValue
				}
			}
			return IntRange(minValue, maxValue)
		}
	}

}